configuration_schemas/v4/server_config.json.in

```diff
@@ -3,15 +3,50 @@
     "$schema": "https://json-schema.org/draft/2020-12/schema",
     "type": "object",
     "properties": {
+        "advanced_settings": {
+            "type": "object",
+            "properties": {
+                "default_number_of_transfer_threads": {"type": "integer"},
+                "default_temporary_password_lifetime_in_seconds": {"type": "integer"},
+                "delay_rule_executors": {
+                    "type": "array",
+                    "items": {"type": "string"},
+                    "uniqueItems": true
+                },
+                "delay_server_sleep_time_in_seconds": {"type": "integer"},
+                "dns_cache": {
+                    "type": "object",
+                    "properties": {
+                        "shared_memory_size_in_bytes": {"type": "integer"},
+                        "eviction_age_in_seconds": {"type": "integer"}
+                    }
+                },
+                "hostname_cache": {
+                    "type": "object",
+                    "properties": {
+                        "shared_memory_size_in_bytes": {"type": "integer"},
+                        "eviction_age_in_seconds": {"type": "integer"}
+                    }
+                },
+                "maximum_size_for_single_buffer_in_megabytes": {"type": "integer"},
+                "maximum_size_of_delay_queue_in_bytes": {"type": "integer"},
+                "maximum_temporary_password_lifetime_in_seconds": {"type": "integer"},
+                "number_of_concurrent_delay_rule_executors": {"type": "integer"},
+                "stacktrace_file_processor_sleep_time_in_seconds": {"type": "integer"},
+                "transfer_buffer_size_for_parallel_transfer_in_megabytes": {"type": "integer"},
+                "transfer_chunk_size_for_parallel_transfer_in_megabytes": {"type": "integer"}
+            }
+        },
         "catalog_provider_hosts": {
             "type": "array",
             "items": {"type": "string"},
             "minItems": 1,
             "uniqueItems": true
         },
         "catalog_service_role": {"enum": ["consumer","provider"]},
+        "client_api_allowlist_policy": {"enum": ["disabled", "enforce"]},
```

Comment:
Q. Any thoughts around this being treated as an enumeration?
Q. How do we feel about the word "disabled"?

Before this PR, we only treated this property as a string. Its value only mattered when it was set to "enforce". Other values were equivalent to "not enforce".

Reply:
I think this is fine.   It limits the choices, and is clear.

Only downside to this... if we come up with another 'mode', we'll have to bump the schema version.   But I cannot think of what that would be at the moment.

Reply:
I can revert it to a string. That provides more flexibility if we feel we may need it.

Reply:
it's basically a boolean for whether we want to make the thing 'go'.   So, I like 'disabled' and 'enforce'.  Let's do it.

Reply:
Cool cool.